### DISCUSSION TITLE
How to put labels in the frame that do not move when changing the length of the text?

### DISCUSSION THREAD

**Discussion by hassan-anari** [TomSchimansky/CustomTkinter#2577] [Q&A] [answered]
Hello experienced friends! @Akascape
One of the problems I have is that when I create a label, I can't update it or change its text because it shifts its position on the page, causing confusion. The example code below includes two labels. Both of them are placed on a vertical axis, but since the text of the second label is longer, it extends outside the visible area. This is a problem for which I don't know the solution. I want the starting point of the text to be fixed, and the rest of the sentence to be placed to the left of the text without changing its position.

```python
import customtkinter as ctk
from tkinter import CENTER

root = ctk.CTk()
root.geometry('400x400')

frame = ctk.CTkFrame(root, width=385, height=385)
frame.place(relx=0.5, rely=0.5, anchor=CENTER)

lable_1 = ctk.CTkLabel(frame, text='A sample text', font=ctk.CTkFont(size=25))
lable_1.place(relx=0.25, rely=0.4, anchor=CENTER)

lable_2 = ctk.CTkLabel(frame, text='A sample text with nothing', font=ctk.CTkFont(size=25))
lable_2.place(relx=0.25, rely=0.6, anchor=CENTER)

root.mainloop()
```
Preview:
![image](https://github.com/user-attachments/assets/bb6fa9b3-3185-4b75-a81a-ee917d220f34)

**Comment by Akascape** [comment]
@Nigacatch Sorry for the late reply,
what you can do is change the word wrap thing here, like this:
```python
lable_2._label.config(wrap=200)
```
Also make sure you use the anchor properly in place method.

**Reply by hassan-anari** [reply]
> @Nigacatch Sorry for the late reply, what you can do is change the word wrap thing here, like this:
> 
> ```python
> lable_2._label.config(wrap=200)
> ```
> 
> Also make sure you use the anchor properly in place method.

@Akascape Thank you so much!
By adding the code you gave, the text is vertically fragmented with a wraplength of 200:

![Screenshot 2024-09-17 130611](https://github.com/user-attachments/assets/6324d5ce-abde-4cc0-8c2d-682b3ef01a8c)

This is good progress! But when I increase wraplength to 300, the same result is obtained again:

![Screenshot 2024-09-17 130719](https://github.com/user-attachments/assets/5021ab68-ed7e-4687-b245-637b3656a525)

But I want the text, for example, in the photo below to be placed in the green box and not moved:

![Screenshot 2024-09-17 130611 - Copy](https://github.com/user-attachments/assets/78f913aa-10cf-4490-9573-b1cb568652ba)

Is such a thing possible? Thank you for always answering!

**Reply by Akascape** [reply, marked as answer]
@Nigacatch I recommend you to use grid method in this case, and use anchor parameter in the ctklabel.
